Kernel/Arch/riscv64/PCI/Initializer.cpp

```diff
@@ -147,8 +151,48 @@ void initialize()
         case ControllerCompatible::Unknown:
             VERIFY_NOT_REACHED(); // This should have been rejected earlier
         }
+
+        auto maybe_ranges = node.get_property("ranges"sv);
+        if (maybe_ranges.has_value()) {
+            auto address_cells = node.get_property("#address-cells"sv).value().as<u32>();
+            VERIFY(address_cells == 3); // Additional cell for OpenFirmware PCI address metadata
+            auto size_cells = node.get_property("#size-cells"sv).value().as<u32>();
+            auto stream = maybe_ranges.value().as_stream();
+            while (!stream.is_eof()) {
+                u32 pci_address_metadata = MUST(stream.read_value<BigEndian<u32>>());
+                MUST(stream.discard((address_cells - 1) * sizeof(u32)));
```

Comment:
I didn't notice it in my review, but we should probably at least verify that "h" and "l"  of the Open Firmware PCI address are equal to `mmio_address`, as `ranges` describes the address space mapping between a child and the parent. And those addresses could be different which would mean that we would have to translate them.
https://www.devicetree.org/open-firmware/bindings/pci/pci2_1.pdf, the "h" and "l" field of 2.2.1.1. The other fields should not be set as described in section 12.

Reply:
We currently don't follow the `ranges` property anywhere though. The `soc` node could also have a `ranges` property which we should use to translate all addresses in the children. So I think it's fine to leave a VERIFY there for now.

Reply:
Not sure I understand what you mean by mapping? Where would we interact with the 'PCI addresses'?

Reply:
Oh I see, theortically the PCI address we write in the BAR could be different than the one we access directly. That sounds really weird, not sure when that would happen in real life. Will add an assertion like you suggested

Reply:
The devicetree `ranges` property describes the address translation between a child and its parent bus.
`ranges` properties are an array of (child-bus-address, parent-bus-address, length) triplets.
The child-bus-address for of the pci controller devicetree node ranges are Open Firmare PCI Addresses. These addresses also contain a "child address" ("h" + "l") next to the metadata which is equal to the parent-bus-address on the QEMU virt machine. But that doesn't have to be the case.
If the addresses are different, you need add the difference between the child and parent bus address.